test/widgets/stream_colors_test.dart

```diff
@@ -0,0 +1,510 @@
+import 'package:checks/checks.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:zulip/widgets/stream_colors.dart';
+
+import 'stream_colors_checks.dart';
+
+void main() {
+  group('StreamColorSwatches', () {
+    test('StreamColorSwatchesLight', () {
+      final instance = StreamColorSwatchesLight.instance;
+
+      const base1 = 0xff76ce90;
+      final swatch1 = instance.forBaseColor(base1);
+      check(swatch1).equals(StreamColorSwatch.light(base1));
+      check(instance.forBaseColor(base1)).identicalTo(swatch1);
+
+      const base2 = 0xfffae589;
+      final swatch2 = instance.forBaseColor(base2);
+      check(swatch2).equals(StreamColorSwatch.light(base2));
+      check(instance.forBaseColor(base2)).identicalTo(swatch2);
+      check(instance.forBaseColor(base1)).identicalTo(swatch1);
+    });
+
+    // TODO deduplicate with corresponding StreamColorSwatchesLight test?
+    test('StreamColorSwatchesDark', () {
+      final instance = StreamColorSwatchesDark.instance;
+
+      const base1 = 0xff76ce90;
+      final swatch1 = instance.forBaseColor(base1);
+      check(swatch1).equals(StreamColorSwatch.dark(base1));
+      check(instance.forBaseColor(base1)).identicalTo(swatch1);
+
+      const base2 = 0xfffae589;
+      final swatch2 = instance.forBaseColor(base2);
+      check(swatch2).equals(StreamColorSwatch.dark(base2));
+      check(instance.forBaseColor(base2)).identicalTo(swatch2);
+      check(instance.forBaseColor(base1)).identicalTo(swatch1);
+    });
+
+    group('lerp', () {
+      test('on identical instances', () {
+        check(
+          StreamColorSwatchesLight.instance.lerp(StreamColorSwatchesLight.instance, 0.5)
+        ).identicalTo(StreamColorSwatchesLight.instance);
+
+        check(
+          StreamColorSwatchesDark.instance.lerp(StreamColorSwatchesDark.instance, 0.5)
+        ).identicalTo(StreamColorSwatchesDark.instance);
+      });
+
+      test('from light to dark', () {
+        final instance = StreamColorSwatchesLight.instance
+          .lerp(StreamColorSwatchesDark.instance, 0.4);
+
+        const base1 = 0xff76ce90;
+        final swatch1 = instance.forBaseColor(base1);
+        check(swatch1).equals(StreamColorSwatch.lerp(
+          StreamColorSwatch.light(base1), StreamColorSwatch.dark(base1), 0.4)!);
+        check(instance.forBaseColor(base1)).identicalTo(swatch1);
+
+        const base2 = 0xfffae589;
+        final swatch2 = instance.forBaseColor(base2);
+        check(swatch2).equals(StreamColorSwatch.lerp(
+          StreamColorSwatch.light(base2), StreamColorSwatch.dark(base2), 0.4)!);
+        check(instance.forBaseColor(base2)).identicalTo(swatch2);
+        check(instance.forBaseColor(base1)).identicalTo(swatch1);
+      });
+
+      test('from dark to light', () {
+        final instance = StreamColorSwatchesDark.instance
+          .lerp(StreamColorSwatchesLight.instance, 0.4);
```

Comment:
I think this test is redundant with the one above it — there isn't any logic in the lerping that works differently between dark and light, so these are just two distinct groups of swatches whichever way around they are.